test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java

```diff
@@ -84,29 +84,32 @@ public static void main(String... args) throws Exception {
      */
     static void runTest() throws Exception {
         // invoke p.Test.run() that loads the native library
-        Runnable r = newTestRunnable();
-        r.run();
-
-        // reload the native library by the same class loader
-        r.run();
+        TestLoader testLoader = newTestRunnable();
```

Comment:
I don't think the test needs to be changed so drastically. It should be sufficient to keep `Runnable r` alive until after `r1.run`. In Java 11+ you can call `java.lang.ref.Reference.reachabilityFence(r1)`.

Reply:
Replaced it with `Reference.reachabilityFence()`.

Reply:
Tested 5x in a grinder, which passed.